### ISSUE TITLE
Golang Programming Overview

### ISSUE THREAD

**Issue by lushenle** [lushenle/lushenle.github.io#5]
https://shenle.lu/post/2022/02/golang-programming-overview/ 

 Go language is a programming language initially developed at Google in the year 2007 by Robert Griesemer, Rob Pike, and Ken Thompson. It is a statically-typed language having syntax similar to that of C. It provides garbage collection, type safety, dynamic-typing capability, many advanced built-in types such as variable length arrays and key-value maps. It also provides a rich standard library. The Go programming language was launched in November 2009 and is used in some of the Google’s production systems.

**Comment by murali1999-tech** [comment]
Concurrency is the main thing in golang

**Comment by lushenle** [comment]
> @murali1999-tech
> Concurrency is the main thing in golang

Yes, but I will go into the concurrency aspect here, this article is asking to prepare for learning data structures later. 

It can't be used as an introduction to a go programming language for a group interview.
